test/commands/distribute/releases/add-destination-test.ts

```diff
@@ -0,0 +1,270 @@
+import { expect, use } from "chai";
+import * as Nock from "nock";
+import * as ChaiAsPromised from "chai-as-promised";
+
+use(ChaiAsPromised);
+
+import AddDestinationCommand from "../../../../src/commands/distribute/releases/add-destination";
+import { CommandArgs, CommandFailedResult, ErrorCodes } from "../../../../src/util/commandline";
+
+describe("releases add-destination command", () => {
+  const fakeAppOwner = "fakeAppOwner";
+  const fakeAppName = "fakeAppName";
+  const fakeAppIdentifier = `${fakeAppOwner}/${fakeAppName}`;
+  const fakeToken = "c1o3d3e7";
+  const fakeReleaseId = "1";
+  const fakeDistributionGroupName = "fakeDistributionGroupName";
+  /* tslint:disable-next-line:no-http-string */
+  const fakeHost = "http://localhost:1700";
+
+  let nockScope: Nock.Scope;
+
+  before(() => {
+    Nock.disableNetConnect();
+  });
+
+  beforeEach(() => {
+    nockScope = Nock(fakeHost);
+  });
+
+  afterEach(() => {
+    Nock.cleanAll();
+  });
+
+  after(() => {
+    Nock.enableNetConnect();
+  });
+
+  describe("validate input parameters", () => {
+    describe("when the release id is not a number", () => {
+      it("reports the command as failed", async () => {
+        const command = new AddDestinationCommand(getCommandArgs(["--release-id", "lol", "--type", "group", "--destination", fakeDistributionGroupName]));
+        const result: CommandFailedResult = await command.execute() as CommandFailedResult;
+
+        expect(result.succeeded).to.be.false;
+        expect(result.errorCode).to.eql(ErrorCodes.InvalidParameter);
+        expect(result.errorMessage).to.eql("lol is not a valid release id");
+      });
+    });
+
+    describe("when an invalid destination type is provided", () => {
+      it("reports the command as failed", async () => {
+        const command = new AddDestinationCommand(getCommandArgs(["--release-id", fakeReleaseId, "--type", "not-a-type", "--destination", fakeDistributionGroupName]));
+        const result: CommandFailedResult = await command.execute() as CommandFailedResult;
+
+        expect(result.succeeded).to.be.false;
+        expect(result.errorCode).to.eql(ErrorCodes.InvalidParameter);
+        expect(result.errorMessage).to.eql("not-a-type is not a valid destination type. Available types are: group, tester");
```

Comment:
I would rather check that the `errorMessage` doesn't start with `not-a-type is not a valid destination type. Available types are: ` here in case the list of available types changes in future (save updating the test).